lib/v2/screens/app/app.dart

```diff
@@ -80,53 +81,69 @@ class _AppState extends State<App> with WidgetsBindingObserver {
-                                const Positioned(right: 3, top: 3, child: NotificationBadge())
-                            ],
-                          ),
-                          label: state.index == i.index ? i.title : '',
+    _navigatorKey = NavigationService.of(context).appNavigatorKey;
+    return BlocProvider(
+      create: (context) => AppBloc(),
+      child: BlocListener<AppBloc, AppState>(
+        listenWhen: (previous, current) => previous.index != current.index,
+        listener: (_, state) => _pageController.jumpToPage(state.index),
+        child: Scaffold(
+          body: PageView(
+            controller: _pageController,
+            physics: const NeverScrollableScrollPhysics(),
+            children: _appScreenItems.map((i) => i.screen).toList(),
+          ),
+          bottomNavigationBar: BlocBuilder<AppBloc, AppState>(
+            builder: (context, state) {
+              return Container(
+                decoration: const BoxDecoration(border: Border(top: BorderSide(color: AppColors.white, width: 0.2))),
+                child: BottomNavigationBar(
+                  currentIndex: state.index,
+                  onTap: (index) => BlocProvider.of<AppBloc>(context).add(BottomBarTapped(index: index)),
+                  selectedLabelStyle: Theme.of(context).textTheme.subtitle3,
+                  unselectedLabelStyle: Theme.of(context).textTheme.subtitle3,
+                  selectedItemColor: AppColors.white,
+                  items: [
+                    for (var i in _appScreenItems)
+                      BottomNavigationBarItem(
+                        activeIcon:
+                            Padding(padding: const EdgeInsets.only(bottom: 4), child: SvgPicture.asset(i.iconSelected)),
+                        icon: Stack(
+                          children: [
+                            SvgPicture.asset(i.icon),
+                            if (state.hasNotification && i.index == 2)
+                              const Positioned(right: 3, top: 3, child: NotificationBadge())
+                          ],
                         ),
-                    ],
-                  ),
-                );
-              },
-            ),
+                        label: state.index == i.index ? i.title : '',
+                      ),
+                  ],
+                ),
+              );
+            },
           ),
         ),
       ),
     );
   }
 }
+
+extension NavigatorStateExtension on NavigatorState {
+  /// Navigate only if the new route is not the same as the current one
+  void pushNamedIfNotCurrent(String routeName, {Object? arguments}) {
+    if (!isCurrent(routeName)) {
+      pushNamed(routeName, arguments: arguments);
+    }
+  }
+
+  bool isCurrent(String routeName) {
+    bool isCurrent = false;
+    popUntil((route) {
+      if (route.settings.name == routeName) {
+        isCurrent = true;
+      }
+      return true;
+    });
+    return isCurrent;
+  }
+}
```

Comment:
Is there a file or folder where I can put this ???

Reply:
hum. cant think of a place . 
leave here, we can move later